src/android/GeofenceJsEvent.java

```diff
@@ -0,0 +1,54 @@
+package com.cowbell.cordova.geofence;
+
+import android.app.Activity;
+import android.util.Log;
+
+import org.apache.cordova.CordovaWebView;
+
+import java.lang.ref.WeakReference;
+import java.util.List;
+
+public class GeofenceJsEvent {
+    public static final String TAG = "GeofencePlugin";
+    public static WeakReference<CordovaWebView> webView = null;
+
+    public static void onTransitionReceived(List<GeoNotification> notifications) {
+        Log.d(TAG, "Transition Event Received!");
+        String js = "setTimeout('geofence.onTransitionReceived("
+                + Gson.get().toJson(notifications) + ")',0)";
+        sendJavascript(js);
+    }
+
+    public static void onNotificationClicked(String data) {
+        if (data != null) {
+            String js = "setTimeout('geofence.onNotificationClicked(" + data + ")', 100)";
```

Comment:
Nice way to share results with our JS code!
If we're using an easier platform than Cordova I would've suggested sending the payload with the notification and read it again once the user clicked it